sql/rowexec/rel.go

```diff
@@ -532,6 +535,27 @@ func (b *BaseBuilder) populateMax1Results(ctx *sql.Context, n *plan.Max1Row, row
 	return nil
 }
 
+func validateOutfile(dir interface{}, fileStr string) error {
+	if dir == nil || dir == "" {
+		return nil
+	}
+	if _, relErr := filepath.Rel(dir.(string), fileStr); relErr != nil {
+		return sql.ErrSecureFilePriv.New()
+	}
+	return nil
+}
+
+func createIfNotExists(fileStr string) (*os.File, error) {
+	if _, fErr := os.Stat(fileStr); fErr == nil {
+		return nil, sql.ErrFileExists.New(fileStr)
+	}
+	file, fileErr := os.Create(fileStr)
+	if fileErr != nil {
+		return nil, fileErr
+	}
```

Comment:
This races on the file being created after we stat the path. Instead, we should use `os.OpenFile(...)` and `os.O_EXCL` to create the file:

```go
file, fileErr := os.OpenFile(fileStr, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0640)
if fileErr != nil {
    return nil, fileErr
}
```

or something like that...